+cowsay

nakajima opened the pull request:
Adds cowsay.

nakajima committed:
add cowsay

nakajima committed:
update

nakajima committed:
cleanup

nakajima committed:
fix everything

nakajima committed:
try install missing perl module for linux

nakajima committed:
add perl

nakajima committed:
remove maybe unnecessary test dep

jhheider commented on projects/github.com/cowsay-org/cowsay/package.yml:
```diff
@@ -0,0 +1,29 @@
+
+platforms:
+  - darwin
+  - linux/x86-64
```
is this intentional? we should either have a comment as to why it doesn't work on linux/aarch64, or remove this stanza if it does.

nakajima replied:
Nope, I think that's just what was there in the generated file.

jhheider commented on projects/github.com/cowsay-org/cowsay/package.yml:
```diff
@@ -0,0 +1,29 @@
+
+build:
+  script:
+    - run: cpan File::Find
```
we should use `cpanmin` to install it in the prefix. see https://github.com/pkgxdev/pantry/blob/main/projects/gnu.org/help2man/package.yml for example.

jhheider commented on projects/github.com/cowsay-org/cowsay/package.yml:
```diff
@@ -0,0 +1,29 @@
+distributable:
+  url: git+https://github.com/cowsay-org/cowsay.git
+  ref: ${{version.tag}}
```
is it the case that using the packaged sources doesn't work? i'm fine if that's the case, but i'd be surprised for a perl package.

nakajima replied:
Nope, updated.

jhheider commented on projects/github.com/cowsay-org/cowsay/package.yml:
```diff
@@ -0,0 +1,29 @@
+    - make install prefix={{prefix}}
+
+provides:
+  - cowsay
```
this should be `- bin/cowsay`

nakajima committed:
respond to feedback

nakajima committed:
update dep

nakajima commented:
@jhheider Thanks for the review, it's my first PR here so I still don't quite have the lay of the land. I think I took care of everything!

jhheider committed:
style fixes

safer to use `{{version.tag}}`, since people aren't the most consistent about keeping or remove `v` from their tags; cpanmin shouldn't be needed at runtime, and the rest is just tweaks

jhheider committed:
needs perl

jhheider reviewed (approved):
looks good!
